lib/resources/df.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+# author: Jonathan Morley
+
+module Inspec::Resources
+  class DfResource < Inspec.resource(1)
+    name 'df'
+    desc 'Use the df InSpec resource to test file system disk space usage'
+    example "
+      describe df('/') do
+        its('size') { should be >= 32000 }
+      end
+    "
+    attr_reader :partition
+
+    def initialize(partition)
+      @partition = partition
+      @cmd = inspec.command("df #{@partition} --output=size | sed \"/blocks/d; s/ *//g\"")
```

Comment:
`--output=size` is not a globally available flag. For example, it doesn't work on macOS or FreeBSD, to my knowledge.

We need to use the new `supports` keyword here to limit this resource to only executing on the operation systems on which this is supported. I would recommend we start with the "linux" family to start. @jquick, could you chime in here and help @tarcinil with the proper `supports` call that will accomplish this?

Reply:
I was unable to find an existing example of `supports` as it pertained to resources.

I used a pre-existing pattern of `inspec.os.linux?` which appears to depend on chef/train in the background.

Reply:
I have found the `supports`/`supported` functions from `lib/inspec/plugins/resource.rb` and `lib/resources/platform.rb, but I am unable to find documentation that would allow me to implement it.  I await implementation details from @jquick.

Reply:
Hey @tarcinil. You are correct that the documentation is lacking for the new supports features for resources, this will be addressed. This was implemented via https://github.com/chef/inspec/issues/1661 (section 4). For this resource you will want to add something like:
`supports os_family: 'linux'`
After you set the 'name' on line 3 in your new resource. This will skip any tests using this resource unless the platform is a linux base.